### DISCUSSION TITLE
Is PK automagically create for fields with name ID or UUID?

### DISCUSSION THREAD

**Discussion by frederikhors** [uptrace/bun#250] [Q&A]
I'm coming from the amazing `go-pg`.

I am writing this to understand if I have understood correctly.

With `go-pg` the below struct gives automagically PK and autoincrement (with sequences) in Postgresql:

```go
type Genre struct {
  ID int
}
```

Instead with `bun` I need to write this:

```go
type Genre struct {
  ID int `bun:",pk,autoincrement"`
}
```

Is this correct?

This sounds strange to me because in this page you say:

```
bun:",pk"	Marks column as a primary key. Multiple primary keys are supported.
bun:",nopk"	Not a primary key. Useful for columns like id and uuid if they are not primary keys.
```

> bun:",nopk"	Not a primary key. Useful for columns like id and uuid if they are not primary keys.

Why?

Is `pk` automagically create for field with name ID or UUID?

**Comment by vmihailenco** [comment]
```go
type Genre struct {
  ID int
}
```

Should be working, e.g. see https://github.com/uptrace/bun/blob/master/internal/dbtest/orm_test.go#L432-L441

**Reply by frederikhors** [reply]
I created an issue for this: https://github.com/uptrace/bun/issues/252.
